packages/angular/projects/clr-angular/src/forms/common/if-success/if-success.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Directive, Input, Optional, TemplateRef, ViewContainerRef } from '@angular/core';
+import { NgControl } from '@angular/forms';
+import { NgControlService } from '../providers/ng-control.service';
+import { IfSuccessService } from './if-success.service';
+import { RxSubscription } from '../../../utils/rx/rx-subscriptions';
+
+@Directive({ selector: '[clrIfSuccess]' })
+export class ClrIfSuccess {
+  private subscriptions = new RxSubscription();
+  private displayed = false;
+  private control: NgControl;
+
+  @Input('clrIfSuccess') success: string;
+
+  constructor(
+    @Optional() private ifSuccessService: IfSuccessService,
+    @Optional() private ngControlService: NgControlService,
+    private template: TemplateRef<any>,
+    private container: ViewContainerRef
+  ) {
+    if (!this.ifSuccessService) {
+      throw new Error('clrIfSuccess can only be used within a form control container element like clr-input-container');
+    }
+
+    this.subscriptions.subscribe = this.ngControlService.controlChanges.subscribe(control => {
+      this.control = control;
+    });
+
+    this.subscriptions.subscribe = this.ifSuccessService.statusChanges.subscribe(state => {
+      this.displaySuccess(state);
+    });
+  }
+
+  ngOnDestroy() {
+    this.subscriptions.unsubscribe();
+  }
+
+  private displaySuccess(valid: boolean) {
+    if (valid && !this.displayed) {
+      this.container.createEmbeddedView(this.template, { success: !this.control.getError(this.success) });
```

Comment:
Do we need to pass this context object? Users will always see `true` this success value.

Reply:
I think that there was a reason for that.. will have to check before I could give an answer ...

Reply:
We don't need it - reworked